src_erl/Communication_Layer/http_Nerlserver/src/Client/clientStatem.erl

```diff
@@ -114,16 +113,43 @@ waitforWorkers(cast, In = {stateChange,WorkerName,MissedBatchesCount}, State = #
 waitforWorkers(cast, In = {NewState}, State = #client_statem_state{myName = MyName, etsRef = EtsRef}) ->
   ets:update_counter(EtsRef, msgCounter, 1),
   ets:update_counter(EtsRef, infoIn, nerl_tools:calculate_size(In)),
-  % ?LOG_INFO("~p in waiting going to state ~p~n",[MyName, State]),
+  %% ?LOG_INFO("~p in waiting going to state ~p~n",[MyName, State]),
   Workers = ets:lookup_element(EtsRef, workersNames, ?ETS_KV_VAL_IDX),
   cast_message_to_workers(EtsRef, {NewState}),
   {next_state, waitforWorkers, State#client_statem_state{nextState = NewState, waitforWorkers = Workers}};
 
+waitforWorkers(cast, In ={worker_kill , Body}, State = #client_statem_state{waitforWorkers = WaitforWorkers,etsRef = EtsRef}) ->
+  %got kill command for a worker, kiil and take off waitforWorkers list
+  ets:update_counter(EtsRef, msgCounter, 1),
+  ets:update_counter(EtsRef, infoIn, nerl_tools:calculate_size(In)),
+  {_, WorkerName} = binary_to_term(Body),
+  Pid = ets:lookup_element(EtsRef, WorkerName, ?WORKER_PID_IDX),
+  gen_statem:stop(Pid, shutdown , infinity),
+  NewWaitforWorkers = WaitforWorkers -- [WorkerName],
+  {next_state, waitforWorkers, State#client_statem_state{waitforWorkers = NewWaitforWorkers,etsRef = EtsRef}};
+
 waitforWorkers(cast, EventContent, State = #client_statem_state{etsRef = EtsRef}) ->
   ets:update_counter(EtsRef, msgCounter, 1),
   ets:update_counter(EtsRef, infoIn, nerl_tools:calculate_size(EventContent)),
-  ?LOG_WARNING("client waitforWorkers ignored!!!:  ~p ~n",[EventContent]),
-  {next_state, waitforWorkers, State}.
+  ?LOG_WARNING("client waitforWorkers ignored!!!:  ~p~n",[EventContent]),
+  {next_state, waitforWorkers, State};
+
+waitforWorkers(info, EventContent, State = #client_statem_state{myName = MyName,waitforWorkers = WaitforWorkers ,etsRef = EtsRef}) ->
+  ets:update_counter(EtsRef, msgCounter, 1),
+  ets:update_counter(EtsRef, infoIn, nerl_tools:calculate_size(EventContent)),
+  case EventContent of
+    {'DOWN',_Ref,process,Pid,_Reason}->                                     %worker down
+      [[WorkerName]] = ets:match(EtsRef,{'$1',Pid,'_','_','_'}),
+      NewWaitforWorkers = WaitforWorkers--[WorkerName],
+      %report worker down to main server
+      delete_worker(EtsRef,MyName,WorkerName),
+      {next_state, waitforWorkers, State#client_statem_state{waitforWorkers = NewWaitforWorkers,etsRef = EtsRef}};           %delete worker from ets so client will not wait for it in the future 
+        
+    _Any-> ?LOG_INFO("client ~p got unexpected info: ~p~n",[MyName, EventContent]),
```

Comment:
What is the term info?
Please stick with our terminology.
There are messages. info is too amorphous.